Packs/PrismaCloudCompute/Integrations/PaloAltoNetworks_PrismaCloudCompute/PaloAltoNetworks_PrismaCloudCompute.py

```diff
@@ -253,9 +268,522 @@ def fetch_incidents(client):
     return incidents
 
 
+def validate_limit_and_offset(func):
+    """
+    Decorator to validate that the limit and offset in the command arguments are correct.
+    The maximum objects that can be returned is 50.
+    """
+    def wrapper(*args, **kwargs):
+
+        # maximum of objects per request
+        api_limit_call = 50
+
+        command_args = kwargs.get('args')
+        offset = arg_to_number(arg=command_args.get('offset'), arg_name='offset')  # type:ignore
+        if offset < 0:  # type:ignore
+            offset = 0
+        command_args['offset'] = offset  # type:ignore
+
+        limit = arg_to_number(arg=command_args.get('limit'), arg_name='limit')  # type:ignore
+        if limit > api_limit_call:  # type:ignore
+            limit = api_limit_call  # type:ignore
+        command_args['limit'] = limit  # type:ignore
+
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+def parse_date_string_format(
+    date_string: str, date_string_format: str = '%Y-%m-%dT%H:%M:%S.%fZ', new_format: str = "%B %d, %Y %H:%M:%S %p"
+) -> str:
+    """
+    Parses a date string format to a different date string format.
+
+    Args:
+        date_string (str): the date in string representation.
+        date_string_format (str): the current format of the string date.
+        new_format (str): the new requested format for the date string.
+
+    Returns:
+        str: date as a new format, in case of a failure returns the original date string.
+    """
+    try:
+        return parse_date_string(date_string=date_string, date_format=date_string_format).strftime(new_format)
+    except ValueError:
+        return date_string
+
+
+def perform_api_request(
+    client: PrismaCloudComputeClient, url_suffix: str, args: dict = None, method: str = 'GET', json_data: dict = None
```

Comment:
type hint of arg with default value of `None` should be `Optional`

Reply:
removed that function